### PR TITLE
[TOO-326] Windows papercuts

### PR DESCRIPTION
**Pull request by jottakka**
> [!NOTE]
> **Medium Risk**
> Touches authentication/login flow, credentials-file permissions, and subprocess lifecycle behavior across platforms; while mostly defensive, regressions could impact login or process management on Windows/macOS runners.
> 
> **Overview**
> Improves Windows/cross-platform reliability across the CLI and MCP server: OAuth login now binds the callback server to `127.0.0.1`, avoids slow loopback reverse-DNS, adds a configurable callback timeout (`--timeout` + env default), and opens URLs via a Windows-friendly `_open_browser` to avoid flashing console windows.
> 
> Centralizes CLI output via a shared `console` that forces UTF-8 on Windows, standardizes UTF-8 file reads/writes throughout, tightens credentials-file permissions on Windows using `icacls`, and adds shared Windows subprocess helpers for **no-window** process creation and graceful termination (used by `deploy`, MCP reload, and usage-tracking worker).
> 
> Updates client configuration UX/robustness (Windows AppData resolution via `platformdirs`, Cursor config path fallbacks + compatibility writes, overwrite warnings, absolute `uv` path for GUI clients, safer path display) and improves `deploy` child-process handling to avoid pipe-buffer deadlocks while giving better debug-aware error messages.
> 
> Expands CI to run tests on Linux/Windows/macOS, adds a no-auth CLI integration workflow, disables usage tracking in toolkits CI, and adds extensive regression tests for Windows signals, subprocess cleanup, UTF-8, and config-path edge cases; bumps `arcade-core` to `4.4.2` and `arcade-mcp-server` to `1.17.2` (with updated dependency pin).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0fabd8ca1cd647039ba6ddbdf3f7809c330bab9e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: libs/arcade-cli/arcade_cli/authn.py [hunk 1]
```diff
@@ -357,21 +384,52 @@ def __init__(self, state: str, port: int = LOCAL_CALLBACK_PORT):
         self.port = port
         self.httpd: HTTPServer | None = None
         self.result: dict[str, Any] = {}
+        self.result_event = threading.Event()
+        self.ready_event = threading.Event()
 
     def run_server(self) -> None:
```
Comment (jottakka) [inline libs/arcade-cli/arcade_cli/authn.py:426]: To avoid affecting previous behavior it should only be called for the widows flow

File: libs/arcade-cli/arcade_cli/authn.py [hunk 2]
```diff
@@ -496,36 +630,38 @@ def build_coordinator_url(host: str, port: int | None) -> str:
...
         with oauth_callback_server(state) as server:
             # server is running and waiting for callback
             ...
-        # After the with block, the server has received the callback
+        # After the with block, the server has been shut down
     """
-    server = OAuthCallbackServer(state)
-    server_thread = threading.Thread(target=server.run_server)
+    server = OAuthCallbackServer(state, port=port)
+    server_thread = threading.Thread(target=server.run_server, daemon=True)
```
Comment (jottakka) [inline libs/arcade-cli/arcade_cli/authn.py:686]: how daemon true affects it ?

File: libs/arcade-cli/arcade_cli/configure.py [hunk 3]
```diff
@@ -53,6 +52,73 @@ def get_windows_username() -> str | None:
...
+    """Resolve Windows roaming AppData via optional ``platformdirs``.
+
+    ``platformdirs`` is the de-facto standard Python library for OS-specific
+    user directory resolution. We import it lazily and treat it as optional so
+    existing environments keep working without new hard dependencies.
+    """
+    try:
+        from platformdirs import user_data_dir
+    except ImportError:
+        return None
```
Comment (jottakka) [inline libs/arcade-cli/arcade_cli/configure.py]: for windows here should return some info saying that this install is required instead of silent fail

File: libs/arcade-cli/arcade_cli/configure.py [hunk 4]
```diff
@@ -53,6 +52,73 @@ def get_windows_username() -> str | None:
...
+
+    try:
+        candidate = user_data_dir(appname=None, appauthor=False, roaming=True)
+    except TypeError:
+        # Compatibility with older signatures that rely on positional args.
```
Comment (jottakka) [inline libs/arcade-cli/arcade_cli/configure.py]: add a debug log here

File: libs/arcade-cli/arcade_cli/configure.py [hunk 5]
```diff
@@ -53,6 +52,73 @@ def get_windows_username() -> str | None:
...
+        candidate = user_data_dir(None, False, roaming=True)
+
+    if not candidate:
+        return None
+    return Path(candidate)
+
+
+def _resolve_windows_appdata() -> Path:
+    """Resolve the Windows roaming AppData directory with fallbacks."""
```
Comment (jottakka) [inline libs/arcade-cli/arcade_cli/configure.py]: are all those fallbacks really needed ? lets stick with one path if it is works always.

File: libs/arcade-cli/arcade_cli/deploy.py [hunk 6]
```diff
@@ -362,6 +367,48 @@ def exclude_filter(tarinfo: tarfile.TarInfo) -> tarfile.TarInfo | None:
...
+                exc,
+            )
+    try:
+        process.terminate()
+    except OSError as exc:
+        logger.debug("terminate() failed during graceful shutdown: %s", exc)
+
+
+def _resolve_server_process_stdio(debug: bool) -> tuple[int | None, int | None]:
+    """Choose stdio strategy for the temporary validation server process.
```
Comment (jottakka) [inline libs/arcade-cli/arcade_cli/deploy.py]: explain this better

File: libs/arcade-core/arcade_core/config_model.py [hunk 7]
```diff
@@ -1,11 +1,43 @@
...
 from typing import Any
 
 import yaml
 from pydantic import BaseModel, ConfigDict, ValidationError
 
+logger = logging.getLogger(__name__)
+
+
+def _set_windows_owner_acl(config_file_path: Path) -> None:
+    """Apply owner-only style ACLs on Windows using ``icacls``.
```
Comment (jottakka) [inline libs/arcade-core/arcade_core/config_model.py]: add better comments here pls

File: libs/arcade-core/arcade_core/config_model.py [hunk 8]
```diff
@@ -230,7 +262,18 @@ def save_to_file(self) -> None:
 
         # Convert to dict, excluding None values for cleaner output
         data = {"cloud": self.model_dump(exclude_none=True, mode="json")}
-        config_file_path.write_text(yaml.dump(data, default_flow_style=False))
+        config_file_path.write_text(yaml.dump(data, default_flow_style=False), encoding="utf-8")
 
-        # Set restrictive permissions (owner read/write only)
-        config_file_path.chmod(0o600)
+        # Set restrictive permissions (owner read/write only).
+        # On Windows, prefer ACL hardening via icacls.
```
Comment (jottakka) [inline libs/arcade-core/arcade_core/config_model.py]: better explanation of what it does and what it means

File: .pre-commit-config.yaml [hunk 9]
```diff
@@ -17,6 +17,6 @@ repos:
     hooks:
       - id: ruff
         args: [--fix]
-        exclude: "(.*/templates/.*|libs/tests/.*)"
```
Comment (jottakka) [inline .pre-commit-config.yaml:20]: wouldnt this change what it is being ignored ? pls check

File: pyproject.toml [hunk 10]
```diff
@@ -42,6 +43,14 @@ all = [
     "numpy>=2.0.0",
     "scikit-learn>=1.5.0",
     "pytz>=2024.1",
+    # platform-aware user directories (optional Windows path resolution)
+    "platformdirs>=4.3.6",
+    # mcp server
+    "arcade-mcp-server>=1.15.3,<2.0.0",
```
Comment (EricGustin) [inline pyproject.toml]: this is already included in the required dependencies, so not needed here as an optional dependency

File: pyproject.toml [hunk 11]
```diff
@@ -42,6 +43,14 @@ all = [
...
+    # serve
+    "arcade-serve>=3.2.0,<4.0.0",
+    # tdk
+    "arcade-tdk>=3.4.0,<4.0.0",
```
Comment (EricGustin) [inline pyproject.toml]: arcade-serve and arcade-tdk are transient dependencies of arcade-mcp-server (a required dep), so they don't need to be here

File: pyproject.toml [hunk 12]
```diff
@@ -154,3 +167,4 @@ ignore = ["E501", "S105"]
 
 [tool.ruff.lint.per-file-ignores]
 "__init__.py" = ["F401"]
+"tests/**/*.py" = ["C901"]
```
Comment (EricGustin) [inline pyproject.toml]: We should reduce the cyclomatic complexity of the test cases instead of ignoring the problem

File: libs/arcade-cli/arcade_cli/authn.py [hunk 13]
```diff
@@ -45,11 +52,22 @@ def _render_template(template_name: str, **context: Any) -> bytes:
...
+def _get_default_oauth_timeout_seconds() -> int:
+    value = os.environ.get("ARCADE_LOGIN_TIMEOUT_SECONDS", str(DEFAULT_OAUTH_TIMEOUT_SECONDS))
+    try:
+        parsed = int(value)
+        return parsed if parsed > 0 else DEFAULT_OAUTH_TIMEOUT_SECONDS  # noqa: TRY300
+    except ValueError:
+        return DEFAULT_OAUTH_TIMEOUT_SECONDS
+
+
+DEFAULT_OAUTH_TIMEOUT_SECONDS = _get_default_oauth_timeout_seconds()
```
[the remark this answers is not in the document]
Comment (EricGustin) [reply libs/arcade-cli/arcade_cli/authn.py:75]: please address this

File: libs/arcade-cli/arcade_cli/authn.py [hunk 14]
```diff
@@ -562,21 +732,39 @@ def status(msg: str) -> None:
...
+            # Step 4: Generate authorization URL and open browser
+            auth_url, code_verifier = generate_authorization_url(
+                oauth_client, cli_config, redirect_uri, state
+            )
+
+            # Always print the auth link first so users can click/copy it
+            # immediately, even if automatic browser launch fails.
+            status("Use this authorization link if needed:\n" f"{auth_url}")
+            status("Opening a browser to log you in...")
+            browser_opened = _open_browser(auth_url)
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/authn.py]: Were you running into issues with the previous logic which only printed the auth link if browser opening failed? I'm of the perspective that we should only print the auth link if opening the browser fails to prevent polluting the output of `arcade login`

File: libs/arcade-cli/arcade_cli/deploy.py [hunk 15]
```diff
@@ -362,6 +367,57 @@ def exclude_filter(tarinfo: tarfile.TarInfo) -> tarfile.TarInfo | None:
...
+    On Windows, ``process.terminate()`` calls ``TerminateProcess`` which kills
+    the child immediately.  If the child was started with
+    ``CREATE_NEW_PROCESS_GROUP`` we can send ``CTRL_BREAK_EVENT`` first so
+    Python's default handler raises ``KeyboardInterrupt`` — giving the child
+    a chance to clean up.
+    """
+    if sys.platform == "win32":
+        try:
+            process.send_signal(signal.CTRL_BREAK_EVENT)  # type: ignore[attr-defined]
+            return  # noqa: TRY300
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/deploy.py]: this should be in an `else` after the `except`

File: libs/arcade-cli/arcade_cli/main.py [hunk 16]
```diff
@@ -144,7 +148,16 @@ def logout(
...
         if os.path.exists(CREDENTIALS_FILE_PATH):
-            os.remove(CREDENTIALS_FILE_PATH)
+            try:
+                os.remove(CREDENTIALS_FILE_PATH)
+            except PermissionError:
+                # On Windows, the file may be locked by another process.
+                handle_cli_error(
+                    "Could not remove credentials file — it may be in use by another process. "
+                    "Close other Arcade instances and try again.",
+                    should_exit=False,
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/main.py]: This should be `should_exit=True`

File: libs/arcade-cli/arcade_cli/main.py [hunk 17]
```diff
@@ -163,8 +176,11 @@ def whoami(
 
     try:
         config = Config.load_from_file()
+    except FileNotFoundError:
+        console.print("Not logged in. Run 'arcade login' to authenticate.", style="bold red")
+        return
     except Exception as e:
-        handle_cli_error("Failed to read credentials", e, debug)
+        handle_cli_error("Failed to read credentials", e, debug, should_exit=False)
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/main.py]: Any reason why we're changing this from the default `should_exit-False` to `should_exit=True`? Seems like this should exit

File: libs/arcade-cli/arcade_cli/main.py [hunk 18]
```diff
@@ -311,8 +327,15 @@ def mcp(
...
-        # Execute the command and pass through all output
-        result = subprocess.run(cmd, check=False)
+        # Execute the command and pass through all output.
+        # On Windows, set CREATE_NO_WINDOW to prevent a phantom console
+        # window from appearing (e.g. when an MCP client spawns this
+        # command without an attached console).  The child process still
+        # inherits stdin/stdout/stderr for stdio transport communication.
+        run_kwargs: dict[str, object] = {"check": False}
+        if sys.platform == "win32":
+            run_kwargs["creationflags"] = subprocess.CREATE_NO_WINDOW
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/main.py]: There is a lot of varying "suppress phantom window" logic spread across various parts of this PR. Is it possible to consolidate this logic into a single helper? Something that would help the future developer?

File: libs/arcade-cli/arcade_cli/main.py [hunk 19]
```diff
@@ -943,6 +966,7 @@ def main_callback(
     public_commands = {
         login.__name__,
         logout.__name__,
+        whoami.__name__,
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/main.py]: "whoami" seems like a command that would require the user to be logged in. Why do we add it as a public command here?

File: libs/arcade-cli/arcade_cli/new.py [hunk 20]
```diff
@@ -271,9 +273,14 @@ def create_new_toolkit_minimal(output_directory: str, toolkit_name: str) -> None
 
     try:
         create_package(env, template_directory, toolkit_directory, context, ignore_pattern)
+        console.print("")
         console.print(
             f"[green]Server '{toolkit_name}' created successfully at '{toolkit_directory}'.[/green]"
         )
+        console.print("Next steps:", style="bold")
+        console.print(f"  cd {toolkit_directory / toolkit_name / 'src' / toolkit_name}")
+        console.print("  uv run server.py")
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/new.py]: DX: give them the command for stdio and for http

File: libs/arcade-core/arcade_core/usage/identity.py [hunk 21]
```diff
@@ -45,7 +45,7 @@ def load_or_create(self) -> dict[str, Any]:
 
         if os.path.exists(self.usage_file_path):
             try:
-                with open(self.usage_file_path) as f:
+                with open(self.usage_file_path, encoding="utf-8") as f:
```
Comment (EricGustin) [inline libs/arcade-core/arcade_core/usage/identity.py:48]: what happens if a future developer forgets to explicitly set `encoding="utf-8"` ? We need a linting rule, or a pre-commit hook, or a unit test to prevent a regression here

File: tests/install/README.md [hunk 22]
```diff
@@ -158,7 +192,18 @@ If you see import errors, make sure you're running from the project root and tha
...
 ```
 
+### Optional eval/display test dependencies
+
+Some CLI test files (for eval/display formatting paths) require optional eval
+dependencies such as `openai`, `pytz`, and `numpy`. Install the eval extras
+before running those suites:
+
+```bash
+uv sync --dev --extra evals
```
Comment (EricGustin) [inline tests/install/README.md]: It's no longer `uv sync --dev`, but instead it's `uv sync --extra dev`. Please update in other locations too

File: tests/integration/mcp_protocol_smoke.py [hunk 23]
```diff
@@ -0,0 +1,670 @@
+#!/usr/bin/env python3
```
Comment (EricGustin) [inline tests/integration/mcp_protocol_smoke.py:1]: Does this differ much from the existing https://github.com/ArcadeAI/arcade-mcp/blob/main/libs/tests/arcade_mcp_server/integration/test_end_to_end.py ? Seems like it's testing a lot of the same stuff and duplicating a lot of the boilerplate.

File: pyproject.toml [hunk 24]
```diff
@@ -42,6 +43,14 @@ all = [
     "numpy>=2.0.0",
     "scikit-learn>=1.5.0",
     "pytz>=2024.1",
+    # platform-aware user directories (optional Windows path resolution)
+    "platformdirs>=4.3.6",
```
Comment (EricGustin) [inline pyproject.toml]: this is already included in the required dependencies, so not needed here as an optional dependency

File: .github/workflows/test-install.yml [hunk 25]
```diff
@@ -40,9 +40,19 @@ jobs:
                   uv pip install -e .
 
             - name: Run installation test
+              env:
+                  ARCADE_USAGE_TRACKING: "0"
```
Comment (EricGustin) [inline .github/workflows/test-install.yml]: Consider adding something like [this](https://github.com/ArcadeAI/arcade-mcp/blob/a918eef0371b83cab23b39be487fdf861977516a/libs/tests/conftest.py#L43-L61) to the integration tests

File: .github/workflows/test-install.yml [hunk 26]
```diff
@@ -40,9 +40,19 @@ jobs:
...
               run: |
                   python tests/install/test_install.py
 
+            - name: Restore full test environment
+              run: |
+                  uv sync --extra all --extra dev
+
+            - name: Run CLI unit tests
+              run: |
+                  uv run python -m pytest libs/tests/cli/ -v --tb=short --ignore=libs/tests/cli/test_display.py --ignore=libs/tests/cli/test_main_evals.py --ignore=libs/tests/cli/deploy/
```
Comment (EricGustin) [inline .github/workflows/test-install.yml]: The `.github/workflows/main.yml` workflow already runs the CLI unit tests. What’s the reason we’re running them again in `test-install.yml`? We'd be running CLI unit tests for linux twice.

It looks like `test-install.yml`’s scope has shifted beyond “test installing the CLI,” so the name feels misleading now. I’d prefer we split responsibilities more clearly: one workflow dedicated to unit tests (with an OS matrix), and a separate workflow dedicated to integration tests (also with an OS matrix).

File: .github/workflows/windows-cli-integration-no-auth.yml [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+        strategy:
+            matrix:
+                python-version: ["3.10", "3.11", "3.12"]
+            fail-fast: false
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+
+            - name: Install uv
```
Comment (EricGustin) [inline .github/workflows/windows-cli-integration-no-auth.yml]: we have a github action for setting up the uv env. Maybe it will help. `.github/actions/setup-uv-env`

File: libs/tests/cli/test_windows_subprocess.py [hunk 28]
```diff
@@ -0,0 +1,239 @@
...
+        """Source must use CTRL_BREAK_EVENT for graceful shutdown on Windows."""
+        import arcade_mcp_server.mcp_app as mcp_mod
+        source = inspect.getsource(mcp_mod)
+        assert "CTRL_BREAK_EVENT" in source
+
+    def test_checks_platform(self) -> None:
+        """Source must check sys.platform before setting Windows flags."""
+        import arcade_mcp_server.mcp_app as mcp_mod
+        source = inspect.getsource(mcp_mod)
+        assert 'sys.platform == "win32"' in source or "sys.platform == 'win32'" in source
```
Comment (EricGustin) [inline libs/tests/cli/test_windows_subprocess.py]: testing source code string matching is very brittle. They're not really testing behavior

File: libs/tests/cli/test_cross_platform.py [hunk 29]
```diff
@@ -0,0 +1,1047 @@
...
+  3. Console encoding safety (UnicodeEncodeError prevention)
+  4. Browser opening without CMD flash
+  5. File locking error handling (os.remove, shutil.rmtree)
+  6. Config model UTF-8 + chmod safety
+  7. Path formatting for display
+  8. OAuth callback loopback binding
+  9. Signal handling logging level
+
+All tests run on BOTH Unix and Windows — platform-specific behaviour is
+tested by mocking ``sys.platform``.
```
Comment (EricGustin) [inline libs/tests/cli/test_cross_platform.py]: I'm hesitant to always mock `sys.platform`. Mocking `sys.platform` seems like it should only be done for very specific scenarios (like testing fallback logic or error handling branches). 

What would be better is if ALL of our tests are run in CI across an OS matrix. If we ran all of our tests in an OS matrix, then this entire file could go away (it's mostly just duplicating existing tests)

Comment (EricGustin) [reply libs/tests/cli/test_cross_platform.py]: For tests that are truly only for a specific OS, then just use `pytest.mark.skipif(sys.platform != "win32", reason="...")`

File: pyproject.toml [hunk 30]
```diff
@@ -20,7 +20,7 @@ requires-python = ">=3.10"
 dependencies = [
     # CLI dependencies
     "arcade-mcp-server>=1.17.0,<2.0.0",
-    "arcade-core>=4.4.0,<5.0.0",
+    "arcade-core>=4.4.2,<5.0.0",
```
Comment (EricGustin) [inline pyproject.toml:23]: arcade-core was bumped to 4.4.1 in this PR, but the dependency is requiring at least 4.4.2. This will completely break the CLI because it will fail to resolve when someone tries to install the CLI

**Review by EricGustin — CHANGES_REQUESTED** [review]
Thank you for tackling this. The callback work, binding fix, and smoke tests are great. 

There are a few blockers before this can merge:
1. The version mismatch in the CLI's dependency will break the CLI when released
2. We should be running the full test suite on the OS matrix.
3. The phantom-window suppression pattern is copied many times throughout with the same Windows logic. Someone in the future is going to forget to paste it again. I would like to see shared subprocess helpers.
4. We need a linting rule to prevent encoding=utf-8 regressions in the future
5. The 1000+ line file `test_cross_platform.py` can arguably be deleted since it's nearly entirely duplicated tests

See inline comments for specifics

File: libs/arcade-cli/arcade_cli/authn.py (hunk 14 above)
Comment (jottakka) [reply libs/arcade-cli/arcade_cli/authn.py]: There were times were it openned in other browser that I wasnt logged in, then I had to login and it would skip the oauth flow other times i just wanted to open in other browser session. I believe having it there is kinda important for usability and give users the option to copy and paste it.

File: libs/tests/core/test_dependency_alignment.py
Comment (jottakka) [inline libs/tests/core/test_dependency_alignment.py:1]: @EricGustin need some feedback about this test, should we have something as its own step?

File: .github/workflows/cli-integration-no-auth.yml [hunk 31]
```diff
@@ -0,0 +1,39 @@
...
+        name: CLI Integration No-Auth (${{ matrix.os }}, Python ${{ matrix.python-version }})
+        runs-on: ${{ matrix.os }}
+        timeout-minutes: 35
+        env:
+            ARCADE_USAGE_TRACKING: "0"
+
+        strategy:
+            matrix:
+                os: [ubuntu-latest, windows-latest, macos-latest]
+                python-version: ["3.10", "3.11", "3.12"]
```
Comment (EricGustin) [inline .github/workflows/cli-integration-no-auth.yml:21]: We also support Python 3.13 and 3.14. This would bring this matrix up to 15 jobs. How many jobs are too many jobs (genuine question - I don't know if we have an answer)?

File: .github/workflows/cli-integration-no-auth.yml [hunk 32]
```diff
@@ -0,0 +1,39 @@
...
+              uses: actions/checkout@v4
+
+            - name: Set up environment
+              uses: ./.github/actions/setup-uv-env
+              with:
+                  python-version: ${{ matrix.python-version }}
+
+            - name: Install arcade-mcp from source
+              run: |
+                  uv pip install -e .
```
Comment (EricGustin) [inline .github/workflows/cli-integration-no-auth.yml:35]: Is this step necessary? The previous "Set up environment" step should already do this

File: .github/workflows/test-toolkits.yml [hunk 33]
```diff
@@ -32,11 +35,13 @@ jobs:
 
   test-toolkits:
     needs: setup
-    runs-on: ubuntu-latest
+    name: test-toolkits (${{ matrix.toolkit }}, ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (EricGustin) [inline .github/workflows/test-toolkits.yml:42]: IMO, and feel free to disagree, but I think doing an OS matrix for non-integration tests and especially for toolkit tests is overkill. We're looking at a ~13x price increase for this matrix for tests that are mostly, if not entirely, testing OS-agnostic things.

File: libs/arcade-cli/arcade_cli/new.py [hunk 34]
```diff
@@ -277,9 +286,15 @@ def create_new_toolkit_minimal(output_directory: str, toolkit_name: str) -> None
...
-        console.print("  uv run server.py")
+        server_dir = toolkit_directory / toolkit_name / "src" / toolkit_name
+        console.print("\nNext steps:", style="bold")
+        console.print(f"  cd {server_dir}")
+        console.print("")
+        console.print("  Run with stdio transport (for MCP clients):", style="dim")
+        console.print("    uv run server.py")
+        console.print("")
+        console.print("  Run with HTTP transport (for development/testing):", style="dim")
+        console.print("    uv run server.py --transport http --port 8000")
```
Comment (EricGustin) [inline libs/arcade-cli/arcade_cli/new.py:297]: I missed the first step to change my directory, and ended up getting an error. Not sure if this is just a me problem or if others would run into this. Maybe numbering the steps would help. 

Image

File: .github/workflows/cli-integration-no-auth.yml [hunk 35]
```diff
@@ -0,0 +1,39 @@
...
+        branches: [main]
+    workflow_dispatch:
+
+jobs:
+    cli-integration-no-auth:
+        name: CLI Integration No-Auth (${{ matrix.os }}, Python ${{ matrix.python-version }})
+        runs-on: ${{ matrix.os }}
+        timeout-minutes: 35
+        env:
+            ARCADE_USAGE_TRACKING: "0"
```
Comment (EricGustin) [inline .github/workflows/cli-integration-no-auth.yml:16]: is this needed in CI anymore since you added that function in conftest.py?
